CodeEdit/Features/Editor/Models/Editor.swift

```diff
@@ -35,22 +35,16 @@ final class Editor: ObservableObject, Identifiable {
     }
 
     /// The current offset in the history list.
+    /// When set, updates the ``selectedTab`` to the tab indicated by the offset.
+    /// See the ``historyOffsetDidChange()`` method for more details.
     @Published var historyOffset: Int = 0 {
         didSet {
-            let tab = history[historyOffset]
-
-            if !tabs.contains(tab) {
-                if let temporaryTab, tabs.contains(temporaryTab) {
-                    closeTab(file: temporaryTab.file, fromHistory: true)
-                }
-                temporaryTab = tab
-                openTab(file: tab.file, fromHistory: true)
-            }
-            selectedTab = tab
+            historyOffsetDidChange()
         }
     }
 
-    /// History of tab switching.
+    /// Maintains the list of tabs that have been switched to.
+    /// - Warning: Use the ``addToHistory(_:)`` or ``clearFuture()`` methods to modify this. Do not modify directly.
     @Published var history: Deque<Tab> = []
```

Comment:
It would be awesome to use private(set) or fileprivate(set) here to keep things encapsulated, but since the extension is in a different file, it's not possible...